### ISSUE TITLE
Active and Deactive Nodes

### ISSUE THREAD

**Issue by wienans** [BehaviorTree/Groot2#47]
**Is your feature request related to a problem? Please describe.**
I find my self currently testing Nodes and also different implementations of Nodes, and i don't want to have multiple implementations of the Nodes running at execution of the Tree, and i don't want to delete them for conveniance if they have many In / Outputs i would need to refill if i want to test the other node again.

**Describe the solution you'd like**
It would be nice to simply right click the Node and deactivate it. In Case of Sequences deactivate the whole sequence and so on.
Would be nice if the Node then is visually greyed out or something.

**Describe alternatives you've considered**
I could pack the Node in a Subtree and therefore don't lose the Information if i remove the subtree but that creates quickly deep trees

Groot Version 1.5.2

**Comment by facontidavide** [comment]
To understand better what you want. Do you mean to activate / deactivate them in the context of the Editor.

This features seems simple for leaf nodes, but not that much for Control nodes. If you have a more detailed description of what you have in mind, I can think about it.

**Comment by wienans** [comment]
Yes i mean it in context of the editor and then obviously effecting the execution. 

Let’s imagine this gripper example:
https://www.behaviortree.dev/docs/tutorial-basics/tutorial_01_first_tree

Let’s imagine  I want to change the approaching algorithm of the gripper. With a other node. And let’s say the old approaching node has many parameters so it would take time to get it back into the state it was before, if I would delete the node. 

I would like to simply deactivate it in the editor so the old node so it and its possible sub nodes don’t get executed, it gets skipped. Or if I don’t have a new node the gripper closes and opens again without moving.

Sure there are other solutions to this problem. Using git and resetting changes but if I want to quickly hop back and forth it’s not nice especially with reloading the tree. Or I could youse a conditional block with always true or always false, which works but is more of a hack. From UX perspective I think deactivating the node via the context menu like in many other node based editors would be the best.

One solution which is not viable as it invalidates the tree but would exactly do what I suggest is removing the connection of that one node and having no node as a root of it. So I can quickly reconnect it again if I need it. I see that that might be a problem due to validation and so on but that’s why I suggest the deactivation.

Regarding you concern for control nodes, I am a bit out of this matter, I didn’t work with the library 1.5 years. But I would simply suggest to skip the node and the according sub nodes if the node which is evaluated is deactivated (like the node never existed)
